src/main/java/org/codehaus/mojo/buildhelper/UpToDatePropertySetting.java

```diff
@@ -0,0 +1,139 @@
+package org.codehaus.mojo.buildhelper;
+
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2004, The Codehaus
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * Holds settings for <code>AbstractUpToDateProperty</code> subclasses.
+ *
+ * @author Adrian Price <demonfiddler @ virginmedia.com>
+ * @since 1.12
+ */
+public final class UpToDatePropertySetting
+{
+    /**
+     * A set of source files.
+     */
+    @Parameter
+    private FileSet fileSet;
+
+    /**
+     * The name of the property to set.
+     */
+    @Parameter( required = true )
+    private String name;
+
+    /**
+     * The property value to set if the up-to-date condition is fulfilled.
+     */
+    @Parameter( defaultValue = "true" )
+    private String value = "true";

```

Comment:
Well that's rather what I thought initially; it's certainly what the docs say and it's how I originally had it set up but I'm pretty sure that I had to initialise the field to its default value in order to actually get it working. Are you aware of anything that could interfere with the normal init-to-defaults mechanism?